qubes/events.py

```diff
@@ -115,6 +115,17 @@ def add_handler(cls, event, func):
         # pylint: disable=no-member
         cls.__handlers__[event].add(func)
 
+    @classmethod
+    def remove_handler(cls, event, func):
+        '''Remove event handler from subject's class.
+
+        :param str event: event identificator
+        :param collections.Callable handler: handler callable
+        '''
+
+        # pylint: disable=no-member
+        cls.__handlers__[event].remove(func)
+
```

Comment:
I don't think passing events directly is a good idea - first of all, we need to apply filters on them (for example those returned by mgmt-permission:mgmt.Events event), but also we may want to introduce more events-processing methods, which would behave differently.
The more I think about it, the more I'm convinced that original approach was better (adding remove_handler function).
You objection about `vm.remove_handler` is invalid anyway, because one shouldn't add/remove events on object instance - it will add/remove even for the whole class, as you said. The same apply to `add_handler`. Is it possible to detect such things and raise some exception?
Yes, `remove_handler` should be called on the same class that `add_handler` was called. It's caller responsibility to do that (the same as providing proper event name, a proper callable, and generally using API properly).
I'll add this to the docstring.